app/components/filters/select-multiple-filter.ts

```diff
@@ -29,59 +33,61 @@ export default class SelectMultipleFilterComponent extends FilterComponent<Signa
 
   @action
   async inserted() {
-    if (this.args.queryParam?.includes('+')) {
-      const needles = deserializeArray(this.args.queryParam).map(
-        (queryParam) => {
-          if (!queryParam) return [];
-          return this.getQueryParam(queryParam);
-        }
-      );
+    const searchField = this.args.searchField;
 
-      const searchField = this.args.searchField;
+    if (!this.router.currentRoute) {
+      console.error('Current route is not available');
+      return;
+    }
 
-      const haystack = (await this.args.options) as unknown as GroupedOptions[];
-      const results: Option[] = [];
+    const results: Option[] = [];
+    const haystack = (await this.args.options) as unknown as GroupedOptions[];
```

Comment:
**Question:** Why do we need to override the type here. Can't we just fix the type in Signature?

https://github.com/lblod/frontend-burgernabije-besluitendatabank/blob/945b617459978a3bb578a7a4f2081113d74116c1/app/components/filters/select-multiple-filter.ts#L14-L20

Reply:
@aliokan because if it gets to that point in the function the type of the `options` array is no longer an array of `Record<string, string>` but instead it is now in the format of powerselect's 'grouped options'.
e.g. `options: [
  groupName: string;
  options: Option[];
]
` instead of `options: Option[]`

Reply:
This is a design error. We cannot declare one type and use another one. 
It not directly related to this bug fix, but we should clean that in a follow up (BNB-546).